### ISSUE TITLE
`generateObject` returns empty object when it fails, we need a way to debug it

### ISSUE THREAD

**Issue by baptisteArno** [vercel/ai#1999]
### Feature Description

I see that it also returns `rawResponse` but that only contains `headers` prop which is not useful for debugging what the model responded

**Comment by baptisteArno** [comment]
Are you open to contrib for this? Do you have an implementation in mind?

**Comment by baptisteArno** [comment]
So after debugging it, I figured it sometimes failed because my prompt was not explicitely saying "Generate a JSON representation".

I wonder if `ai` should by default add that bit to the provided prompt.

**Comment by lgrammel** [comment]
Depending on how it's configured, the JSON information should be injected automatically. You can use fetch overrides to get details for now: https://github.com/vercel/ai/blob/main/examples/ai-core/src/generate-text/openai-custom-fetch.ts

**Comment by MentalGear** [comment]
I second this issue. I find debugging `generateObject` quite opaque and results not stable - a full rawResponse is needed.

**Comment by lgrammel** [comment]
There was a bug around temperature settings during object generation that has been fixed with `3.2.0`: https://github.com/vercel/ai/pull/2012

The temp for object generation with tool calls is now always `0` (was `undefined` before). Results with e.g. OpenAI should be better now.

**Comment by MentalGear** [comment]
That's good to know. Can we still expect a full rawResponse though ?

**Comment by OrhanTozan** [comment]
I agree with @MentalGear. We need to be able to see the direct output of the provider API.

**Comment by lgrammel** [comment]
https://github.com/vercel/ai/pull/3073

**Comment by lgrammel** [comment]
Please check out https://sdk.vercel.ai/docs/ai-sdk-core/generating-structured-data#error-handling

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Feature**  
Confidence: **High**

### Rationale
The central request is to expand `generateObject` diagnostics by exposing the provider's full raw response. This is an API/observability enhancement rather than a remaining confirmed failure of promised behavior.

### Evidence
- The reporter says `rawResponse` exposes only headers, making failed `generateObject` calls difficult to debug.
- Multiple commenters explicitly request the full/direct provider response.
- A maintainer identified and fixed a separate temperature bug, while later directing users to error-handling documentation as a workaround.
